tests/xharness/Jenkins.cs

```diff
@@ -2077,7 +2077,14 @@ <li>Toggle visibility
 										log_target = "_self";
 										break;
 									}
-									writer.WriteLine ("<a href='{0}' type='{2}' target='{3}'>{1}</a><br />", LinkEncode (log.FullPath.Substring (LogDirectory.Length + 1)), log.Description, log_type, log_target);
+									if (log.Description == "Build log") {
+										var binlog = log.FullPath.Replace (".txt", ".binlog");
+										var textLink = string.Format ("<a href='{0}' type='{2}' target='{3}'>{1}</a>", LinkEncode (log.FullPath.Substring (LogDirectory.Length + 1)), log.Description, log_type, log_target);
+										var binLink = string.Format ("<a href='{0}' type='{2}' target='{3}' style='display:{4}'>{1}</a><br />", LinkEncode (binlog.Substring (LogDirectory.Length + 1)), "Binlog download", log_type, log_target, test.Building? "none" : "inline");
```

Comment:
Not all `Build log`s have bin logs, so checking if the file exists before adding the link would be nice.